navigator-menu/src/contexts/TipsCalculatorContext.jsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const TipsCalculatorContext = createContext();
+
+const initialState = {
+  bill: 0,
+  percentage: 0,
+  people: 0,
+};
+
+export const TipsCalculatorInfoProvider = ({ children }) => {
+  const [tipsCalc, setTipsCalc] = useState(initialState);
+
+  const bill = (amount) => {
+    setTipsCalc((values) => ({
+      ...values,
```

Comment:
This is not a task for the PR, but check out immerjs/immer for a good solution on how to get rid of this spreading current state shit.